content/assignments/minishell/src/utils.h

```diff
@@ -0,0 +1,29 @@
+#ifndef _UTILS_H
+#define _UTILS_H
+
+#include "../util/parser/parser.h"
+
+
+/* useful macro for handling error codes */
+#define DIE(assertion, call_description) \
+	do { \
+		if (assertion) { \
+			fprintf(stderr, "(%s, %s, %d): ", \
+				__FILE__, __FUNCTION__, __LINE__); \
+			perror(call_description); \
+			exit(EXIT_FAILURE); \
+		} \
+	} while (0)
```

Comment:
Try to use the macro from the common `utils.h` file [1] and change the name of this file to `parser_utils.h`.

[1] https://github.com/open-education-hub/operating-systems/blob/master/content/common/utils/utils.h

Reply:
> It's the same situation that we have with the Makefile system.

Honestly, I don't mind this :))). Duplication sucks because someone sometime will surely forget to propagate some changes. But I see your point that it might be difficult for newcomers to understand it. Keep it as it is, but align the `\`s.